File: app/src/main/java/com/anokmik/tripassistant/author/AuthorActivity.java
package com.anokmik.tripassistant.author;

import android.os.Bundle;
import android.view.View;

import com.anokmik.persistence.model.Author;
import com.anokmik.tripassistant.R;
import com.anokmik.tripassistant.author.login.LoginFragment;
import com.anokmik.tripassistant.author.profile.ProfileFragment;
import com.anokmik.tripassistant.base.BaseActivity;
import com.anokmik.tripassistant.databinding.ActivityAuthorBinding;

public final class AuthorActivity extends BaseActivity<ActivityAuthorBinding> implements AuthorContract.View, View.OnClickListener {

    private final AuthorPresenter authorPresenter;

    public AuthorActivity() {
        this.authorPresenter = new AuthorPresenter(this);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (savedInstanceState == null) {
            authorPresenter.validateAuthor();
        }
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_author;
    }

    @Override
    protected void initBinding(ActivityAuthorBinding binding) {
        setSupportActionBar(binding.toolbar);
        setContainerId(binding.container.getId());
        binding.setNavigationClickListener(this);
    }

    @Override
    public void showLogin() {
        addFragment(LoginFragment.newInstance());
    }

    @Override
    public void showProfile(long authorId) {
        addFragment(ProfileFragment.newInstance(authorId));
    }

    @Override
    public void onClick(View view) {
        onBackPressed();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/author/AuthorActivity.kt
package com.anokmik.tripassistant.author

import android.os.Bundle
import android.view.View

import com.anokmik.persistence.model.Author
import com.anokmik.tripassistant.R
import com.anokmik.tripassistant.author.login.LoginFragment
import com.anokmik.tripassistant.author.profile.ProfileFragment
import com.anokmik.tripassistant.base.BaseActivity
import com.anokmik.tripassistant.databinding.ActivityAuthorBinding

class AuthorActivity : BaseActivity<ActivityAuthorBinding>(), AuthorContract.View, View.OnClickListener {

    private val authorPresenter = AuthorPresenter(this)

    init {
        layoutId = R.layout.activity_author
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (savedInstanceState == null) {
            authorPresenter.validateAuthor()
        }
    }

    override fun initBinding(binding: ActivityAuthorBinding) {
        setSupportActionBar(binding.toolbar)
        containerId = binding.container.id
        binding.navigationClickListener = this
    }

    override fun showLogin() {
        addFragment(LoginFragment.newInstance())
    }

    override fun showProfile(author: Author) {
        addFragment(ProfileFragment.newInstance(author.id))
    }

    override fun onClick(view: View) = onBackPressed()

}